src/main/groovy/org/gradle/android/workarounds/JdkImageWorkaround.groovy

```diff
@@ -86,14 +98,17 @@ class JdkImageWorkaround implements Workaround {
     }
 
     private static void applyToAllAndroidVariantsWithNewVariantApi(Project project, Closure<?> configureVariant) {
-        def androidComponents = project.extensions.findByName("androidComponents")
-        def selector = androidComponents.selector()
-        androidComponents.onVariants(selector.all(), configureVariant)
+        project.plugins.withId("com.android.base") {
+            def androidComponents = project.extensions.findByName("androidComponents")
+            def selector = androidComponents.selector()
+            androidComponents.onVariants(selector.all(), configureVariant)
+        }
     }
 
     static def applyRuntimeClasspathNormalization(Project project) {
         project.normalization { handler ->
             handler.runtimeClasspath {
+                it.ignore '**/java/lang/invoke/**'
```

Comment:
@runningcode I think we need to include this normalization in the workaround.
I know it was discussed here https://github.com/gradle/android-cache-fix-gradle-plugin/issues/341#issuecomment-1234700697 but I wanted to confirm.

In a M1 scenario with AGP 7.4 and different java versions, if we remove this line, the result is:
https://ge.solutions-team.gradle.com/c/gv6iro7iyouvy/xituylgavkwve/task-inputs?expanded=WyJqZ3BseXgydm90d3dpLW9wdGlvbnMuY29tcGlsZXJhcmd1bWVudHByb3ZpZGVycy4kMy5leHRyYWN0ZWRqZGtpbWFnZSJd
We need to apply the normalization manually to fix this issue. So the workaround is not enough(Similar to 7.3.1)
Wdyt?

Reply:
Happy to discuss this again. Even though it is happening in 7.4 just like it was happening in 7.3. I still stand by my comment there and I suggest removing this line for the same reasons. I don't think we should ignore `java/lang/invoke` implicitly across all modules where this plugin is applied.

Reply:
My only preference is to avoid the normalization from the client side, other than that I'm ok removing it...

Reply:
I see both sides but I think normalizing such an important package like `java/lang/invoke` should be done explicitly. Unfortnately, normalization applies to all tasks in the project and not just the one task we care about here.

Reply:
removed https://github.com/gradle/android-cache-fix-gradle-plugin/pull/419/commits/cf863228d1e0276c3ee94de9d9a7fbdae6759f94